src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletOperationsTests.cs

```diff
@@ -1191,6 +1191,159 @@ public async Task GetWalletGeneralInfo()
             generalInfoModel.WalletFilePath.Should().Be(this.fixture.WalletWithFundsFilePath);
         }
 
+        [Fact]
+        public async Task BuildTransactionFromWallet()
+        {
+            // Act.
+            WalletBuildTransactionModel buildTransactionModel = await $"http://localhost:{this.fixture.Node.ApiPort}/api"
+                .AppendPathSegment("wallet/build-transaction")
+                .PostJsonAsync(new BuildTransactionRequest
+                {
+                    WalletName = this.fixture.walletWithFundsName,
+                    AccountName = "account 0",
+                    FeeType = "low",
+                    Password = "123456",
+                    ShuffleOutputs = true,
+                    AllowUnconfirmed = true,
+                    Recipients = new List<RecipientModel> { new RecipientModel {DestinationAddress = "TX725W9ngnnoNuXX6mxvx5iHwS9VEuTa4s", Amount = "1000" } }
+                })
+                .ReceiveJson<WalletBuildTransactionModel>();
+
+            // Assert.
+            buildTransactionModel.Fee.Should().Be(new Money(3780));
+
+            Transaction trx = this.fixture.Node.FullNode.Network.Consensus.ConsensusFactory.CreateTransaction(buildTransactionModel.Hex);
+            trx.Outputs.Should().Contain(o => o.Value == Money.COIN * 1000 && o.ScriptPubKey == BitcoinAddress.Create("TX725W9ngnnoNuXX6mxvx5iHwS9VEuTa4s", this.fixture.Node.FullNode.Network).ScriptPubKey);
+        }
+
+        [Fact]
+        public async Task BuildTransactionWithSelectedInputs()
+        {
+            // Act.
+            WalletBuildTransactionModel buildTransactionModel = await $"http://localhost:{this.fixture.Node.ApiPort}/api"
+                .AppendPathSegment("wallet/build-transaction")
+                .PostJsonAsync(new BuildTransactionRequest
+                {
+                    WalletName = this.fixture.walletWithFundsName,
+                    AccountName = "account 0",
+                    FeeType = "low",
+                    Password = "123456",
+                    ShuffleOutputs = true,
+                    AllowUnconfirmed = true,
+                    Recipients = new List<RecipientModel> { new RecipientModel { DestinationAddress = "TX725W9ngnnoNuXX6mxvx5iHwS9VEuTa4s", Amount = "1000" } },
```

Comment:
Rather than hardcode network-dependent base58 addresses, could this be a uint160 which is converted to base58 based on the test network? We've started doing this in our SC tests as it makes it much easier to paramaterize the test network.

Reply:
ah yeah that's clever. I'll do that.